LinearRegression/Blog/main.tex

```diff
@@ -0,0 +1,235 @@
+\subsection*{Correlation Matrix}
+The correlation coefficient is a statistic that measures linear correlation between two variables. Its value lies between -1 and +1. The absolute value gives the strength of the relationship between the two variables. The larger its magnitude, the stronger their relationship. In general, a correlation coefficient above +0.8 or below -0.8 indicates a strong correlation, and we should take some measures to reduce multicollinearity before we continue.
+
+The pandas corr() function calculates the correlations among all variables in the dataframe. We can visualize the resulting correlation matrix using a heatmap from the seaborn library. The diagonal values should always equal 1 since they are the correlations of a variable with itself.
+\begin{figure}[H]\includegraphics[width=1.1\linewidth]{8}\end{figure}
+\begin{figure}[H]\includegraphics[width=0.9\linewidth]{9}\end{figure}
+\subsection*{VIF Factors}
+VIF factor is an alternative measure for multicollinearity, which ranges from 1 upwards. A rule of thumb used in practice is if a VIF value is greater than 4, then multicollinearity is a problem. We can check VIF factors using the variance\_inflation\_factor() function from the statsmodels package.
+\begin{figure}[H]\includegraphics[width=\linewidth]{10}\end{figure}
+\begin{figure}[H]\centering\includegraphics[width=0.5\linewidth]{11}\end{figure}
+You should see in the correlation matrix that the correlation coefficient between X3 and X6 is $-0.81 < -0.8$, and the VIF factor for X3 is also slightly higher than 4, which is unfavorable. A simple solution is to remove one of the correlated variables from our model. After removing X3 from the list of column names and rerunning the previous code, the new correlation matrix should show no significant multicollinearity, and the VIF factors of all the remaining variables are around 1, so we are in good shape, and can proceed with our regression.
+\begin{figure}[H]\includegraphics[width=0.8\linewidth]{12}\end{figure}
+\begin{figure}[H]\centering\includegraphics[width=0.5\linewidth]{13}\end{figure}
+But wait, you may ask, why did I choose to delete X3 rather than X6? Indeed, my choice was kind of arbitrary here. To remove the redundant variables more carefully, we can consider using stepwise regression to only keep the predictors that lead to the best performance; I will describe backwards stepwise regression later.
+
+Another way to deal with multicollinearity without having to drop your predictors before the analysis is to perform regression with regularization techniques (such as Lasso and Ridge) which I will also describe at the end of the article. Regularization can help you handle multicollinearity, so if you don't want to delete any variables, you may choose to skip the OLS model below and directly jump to the Lasso/Ridge/Elastic Net regression presented in the end.
```

Comment:
Optional, but the phrase "which I will also describe at the end of the article" could be omitted for concision, given that "Lasso" and "Ridge" are already the headings of two sections.